drivers/wifi/nrf700x/zephyr/src/zephyr_wpa_supp_if.c

```diff
@@ -868,4 +889,88 @@ int wifi_nrf_wpa_set_supp_port(void *if_priv, int authorized, char *bssid)
 
 	return wifi_nrf_fmac_chg_sta(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, &chg_sta_info);
 }
+
+int wifi_nrf_wpa_supp_signal_poll(void *if_priv, struct wpa_signal_info *si, unsigned char *bssid)
+{
+	enum wifi_nrf_status status = WIFI_NRF_STATUS_FAIL;
+	struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = NULL;
+	struct wifi_nrf_ctx_zep *rpu_ctx_zep = NULL;
+	struct wifi_nrf_fmac_dev_ctx *fmac_dev_ctx = NULL;
+	struct img_umac_cmd_get_sta *sta_info;
+	int ret = -1;
+	int i = 0;
+
+	if (!if_priv) {
+		LOG_ERR("%s: Invalid params\n", __func__);
+		goto out;
+	}
+
+	vif_ctx_zep = if_priv;
+	rpu_ctx_zep = vif_ctx_zep->rpu_ctx_zep;
+	fmac_dev_ctx = rpu_ctx_zep->rpu_ctx;
+
+	signal_info = si;
+	ret = wifi_nrf_fmac_get_station(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx, bssid);
+
+	k_sem_take(&wait_for_event_sem, K_FOREVER);
+
+	ret = wifi_nrf_fmac_get_interface(rpu_ctx_zep->rpu_ctx, vif_ctx_zep->vif_idx);
+
+	k_sem_take(&wait_for_event_sem, K_FOREVER);
+	signal_info->frequency = vif_ctx_zep->assoc_freq;
+out:
+	return ret;
+}
+
+void wifi_nrf_wpa_supp_event_proc_get_station(void *if_priv,
+					   struct img_umac_event_new_station *info,
+					   unsigned int event_len)
+{
+	struct wifi_nrf_vif_ctx_zep *vif_ctx_zep = NULL;
+
+	vif_ctx_zep = if_priv;
+
+	if (info->sta_info.valid_fields & IMG_STA_INFO_SIGNAL_VALID) {
+		signal_info->current_signal = info->sta_info.signal;
+	}
+
+	if (info->sta_info.valid_fields & IMG_STA_INFO_SIGNAL_AVG_VALID) {
+		signal_info->avg_signal = info->sta_info.signal_avg;
+	} else {
+		signal_info->avg_signal = 0;
```

Comment:
just do a memset 0. and also, 0 might be a valid value for signal, so, we should use some `WPA_INVALID_NOISE`

Reply:
Assigned WPA_INVALID_NOISE in case of signal read failure